packages/amplication-server/src/core/resource/resourceBtm.service.ts

```diff
@@ -0,0 +1,261 @@
+import { Injectable } from "@nestjs/common";
+import { AmplicationError } from "../../errors/AmplicationError";
+import { EnumDataType, PrismaService } from "../../prisma";
+import { INVALID_RESOURCE_ID } from "./resource.service";
+import {
+  BreakTheMonolithPromptInput,
+  BreakTheMonolithPromptOutput,
+  EntityPartial,
+  ResourcePartial,
+} from "./resourceBtm.types";
+import { GptService } from "../gpt/gpt.service";
+import { ConversationTypeKey } from "../gpt/gpt.types";
+import { UserAction } from "../userAction/dto";
+import { EnumUserActionStatus } from "../userAction/types";
+import {
+  BreakServiceToMicroserviceResult,
+  BreakTheMonolithRecommendationsResult,
+} from "./dto/BreakServiceToMicroserviceResult";
+import { UserActionService } from "../userAction/userAction.service";
+import { AiBadFormatResponseError } from "./errors/AiBadFormatResponseError";
+
+@Injectable()
+export class ResourceBtmService {
+  /* eslint-disable @typescript-eslint/naming-convention */
+  private dataTypeMap: Record<keyof typeof EnumDataType, string> = {
+    SingleLineText: "string",
+    MultiLineText: "string",
+    Email: "string",
+    WholeNumber: "int",
+    DateTime: "datetime",
+    DecimalNumber: "float",
+    Lookup: "enum",
+    MultiSelectOptionSet: "enum",
+    OptionSet: "enum",
+    Boolean: "bool",
+    GeographicLocation: "string",
+    Id: "int",
+    CreatedAt: "datetime",
+    UpdatedAt: "datetime",
+    Roles: "string",
+    Username: "string",
+    Password: "string",
+    Json: "string",
```

Comment:
do we need this? please check the results against GPT with or without it and if this is not needed let's get rid of it

Reply:
@yuval-hazaz, Dani played a lot with the prompt (coordinated with Paz), to get the results we have now. While I can't say confidently that this will change the results, I think Dani will have more insight into this, and if we test on a single schema, we might not know (Dani tested more extensively until he reached\ this prompt).
Therefore, I suggest leaving it as-is for now (seeing also that the cost of removing this later is 2 minutes of work), and waiting to consult with Dani when he returns. We can open a ticket to remind us to look into this again and evaluate if we need this mapping or not.